descqa/ColorDistribution.py

```diff
@@ -114,26 +134,36 @@ def run_on_single_catalog(self, catalog_instance, catalog_name, output_dir):
         else:
             possible_lsst_names = (('mag_{}_noagn_lsst', 'mag_true_{}_noagn_lsst')
                                    if self.exclude_agn else ('mag_{}_lsst', 'mag_true_{}_lsst'))
-            possible_names = ('mag_{}_sdss', 'mag_{}_des', 'mag_true_{}_sdss', 'mag_true_{}_des') + possible_lsst_names
+            possible_non_lsst_names = ('mag_{}_sdss', 'mag_{}_des', 'mag_true_{}_sdss', 'mag_true_{}_des')
+            if self.use_lsst:
+                print('Selecting lsst magnitudes if available')
+                possible_names = possible_lsst_names + possible_non_lsst_names
+            else:
+                possible_names = possible_non_lsst_names + possible_lsst_names
 
         labels = {band: catalog_instance.first_available(*(n.format(band) for n in possible_names)) for band in bands}
         labels = {k: v for k, v in labels.items() if v}
 
         if len(labels) < 2:
             return TestResult(skipped=True, summary='magnitudes in mock catalog do not have at least two needed bands.')
-        filters = set((v.rpartition('_')[-1] for v in labels.values()))
+        filters = set((v.split('_')[(-2 if 'z0' in v else -1)] for v in labels.values()))
 
         if len(filters) > 1:
             return TestResult(skipped=True, summary='magnitudes in mock catalog have mixed filters.')
         filter_this = filters.pop()
 
-        labels['redshift'] = 'redshift_true'
-        if not catalog_instance.has_quantity(labels['redshift']):
-            return TestResult(skipped=True, summary='mock catalog does not have redshift.')
+        if self.lightcone:
+            labels['redshift'] = catalog_instance.first_available('redshift_true', 'redshift')
+            if not labels['redshift']:
+                return TestResult(skipped=True, summary='mock catalog does not have redshift.')
 
-        # Load mock catalog data
-        filters = ['{} > {}'.format(labels['redshift'], self.zlo),
-                   '{} < {}'.format(labels['redshift'], self.zhi)]
+            # Load mock catalog data
+            filters = ['{} > {}'.format(labels['redshift'], self.zlo),
+                       '{} < {}'.format(labels['redshift'], self.zhi)]
+        else:
+            filters = None
+            self.redshift = catalog_instance.redshift
```

Comment:
This may cause unintended effect because the instance of the test class will be reused for all catalogs. 
```suggestion
            self.redshift = catalog_instance.redshift
```

Reply:
@yymao Not sure what you were suggesting here, but I realized that I didn't need redshift to be a class variable. It is now local. This will fix the travis issue. Let me know if it doesn't address your concern.